### ISSUE TITLE
PyUnitTestCase.id is redundant and should be removed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1950]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#1950|
|Type|enhancement|
|Created|2006-07-23 06:00:53Z|

The reason it has not been is that it causes doctest tests to start failing, but perhaps doctest should have individual support.

<details><summary>Searchable metadata</summary>

```
trac-id__1950 1950
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1153634453000000 1153634453000000
changetime__1195242065000000 1195242065000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

The branch for [#2898](https://github.com/twisted/twisted/issues/2898) should make this easier to change.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|

(In [21788]) Hey look! I can delete this and the tests still pass.

Turns out that our re-implementation of id() for pyunit TestCase is totally
unnecessary.

Refs [#1950](https://github.com/twisted/twisted/issues/1950).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml commented|
|-|-|

Now being fixed in the branch for [#2898](https://github.com/twisted/twisted/issues/2898).

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set status to `closed`|
|-|-|

(In [21827]) Define `ITestCase` to be the interface that test frameworks expect of a test.

 * Author: jml
 * Reviewers: therve, exarkun
 * Fixes [#2898](https://github.com/twisted/twisted/issues/2898)

`ITestCase` is re-purposed to define what you can expect from a test object.
We use this interface to adapt pyunit tests to things usable in Trial.

In the process we also clarify what's needed for compatibility with doctest
and pyunit.

This branch also fixes [#1950](https://github.com/twisted/twisted/issues/1950).

Refs [#2739](https://github.com/twisted/twisted/issues/2739).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
